pkg/diagnoser/networkoutbound_diagnoser.go

```diff
@@ -51,25 +54,26 @@ func (diagnoser *NetworkOutboundDiagnoser) Diagnose() error {
 	networkOutboundDiagnosticFile := filepath.Join(rootPath, diagnoser.GetName())
 
 	f, err := os.OpenFile(networkOutboundDiagnosticFile, os.O_CREATE|os.O_WRONLY, 0644)
-	defer f.Close()
 	if err != nil {
 		return fmt.Errorf("Fail to open file %s: %+v", networkOutboundDiagnosticFile, err)
 	}
+	defer f.Close()
 
 	outboundDiagnosticData := []networkOutboundDiagnosticDatum{}
 
 	for _, file := range diagnoser.networkOutboundCollector.GetCollectorFiles() {
 		t, err := os.Open(file)
-		defer t.Close()
 		if err != nil {
 			return fmt.Errorf("Fail to open file %s: %+v", file, err)
 		}
+		defer t.Close()
 
 		dataPoint := networkOutboundDiagnosticDatum{HostName: hostName}
 		scanner := bufio.NewScanner(t)
 		for scanner.Scan() {
 			var outboundDatum collector.NetworkOutboundDatum
-			json.Unmarshal([]byte(scanner.Text()), &outboundDatum)
+			err := json.Unmarshal([]byte(scanner.Text()), &outboundDatum)
+			fmt.Println(err)
```

Comment:
I think we use log.Printf instead of fmt
Also if we are having an error, may be we should have a `continue` to move to the next item ?